linter/lints/common.go

```diff
@@ -35,3 +37,50 @@ func GetExtWithOID(exts []pkix.Extension, oid asn1.ObjectIdentifier) *pkix.Exten
 	}
 	return nil
 }
+
+// ReadOptionalASN1BooleanWithTag attempts to read and advance incoming to
+// search for an optional DER-encoded ASN.1 element tagged with the given tag.
+// Unless outExpectedTagPresent or outPresent are nil, it stores whether the tag
+// was found in outExpectedTagPresent and an element with the tag was found in
+// outPresent, otherwise each boolean will take the default value. It reports
+// whether all reads were successful.
+func ReadOptionalASN1BooleanWithTag(incoming *cryptobyte.String, outExpectedTagPresent *bool, outPresent *bool, tag cryptobyte_asn1.Tag, defaultValue bool) bool {
+	if outExpectedTagPresent != nil {
```

Comment:
In https://github.com/letsencrypt/boulder/pull/7064#discussion_r1339204471 we talked about using the `defaultValue` param instead of the `present` idiom, in part because that's what upstream does and in part because it's more convenient.

This doesn't allow us to distinguish between:

 1. a value that's present, with the default value, vs
 2. a value that's absent

But we don't really care because in DER, (1) is forbidden. So even if we happen to get (1), hopefully some other lint would catch the problem "this is BER not DER."

So we should have just one outparam `out *bool`. The overall function signature should be:

```
ReadOptionalASN1BooleanWithTag(incoming *cryptobyte.String, out *bool, tag asn1.Tag, defaultValue bool) bool
```

Reply:
We discussed this in video chat and I understand now.